### ISSUE TITLE
SyntaxError: Expected atom at position 2     \p{Script_Ex 

### ISSUE THREAD

**Issue by satishmane** [babel/babel#8951]
## Bug Report

**Current Behavior**

```
> babel . --out-dir dist

SyntaxError: Expected atom at position 2
    \p{Script_Ex
      ^
    at bail (/home/shm/dpp-query-ws/node_modules/regjsparser/parser.js:1093:13)
    at parseTerm (/home/shm/dpp-query-ws/node_modules/regjsparser/parser.js:486:9)
    at parseAlternative (/home/shm/dpp-query-ws/node_modules/regjsparser/parser.js:457:21)
    at parseDisjunction (/home/shm/dpp-query-ws/node_modules/regjsparser/parser.js:437:16)
    at parse (/home/shm/dpp-query-ws/node_modules/regjsparser/parser.js:1108:18)
    at rewritePattern (/home/shm/dpp-query-ws/node_modules/regexpu-core/rewrite-pattern.js:324:15)
    at PluginPass.RegExpLiteral (/home/shm/dpp-query-ws/node_modules/@babel/plugin-transform-unicode-regex/lib/index.js:50:51)
    at newFn (/home/shm/dpp-query-ws/node_modules/@babel/traverse/lib/visitors.js:193:21)
    at NodePath._call (/home/shm/dpp-query-ws/node_modules/@babel/traverse/lib/path/context.js:53:20)
    at NodePath.call (/home/shm/dpp-query-ws/node_modules/@babel/traverse/lib/path/context.js:40:17)
```

**Input Code**
- REPL or Repo link if applicable:

```js
var your => (code) => here;
```

**Expected behavior/code**
Successful transpiled code

**Babel Configuration (.babelrc, package.json, cli command)**
package.json
```json5
"devDependencies": {
    "@babel/cli": "^7.1.2",
    "@babel/core": "^7.1.2",
    "@babel/node": "^7.0.0",
    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
    "@babel/preset-env": "^7.1.0",
    "@babel/preset-flow": "^7.0.0",
    "nodemon": "^1.18.4"
  }
```
babel.config.js
```js
module.exports = {
  "presets": ["@babel/preset-env","@babel/preset-flow"],
  "plugins": [
    "@babel/plugin-proposal-object-rest-spread"
  ]
}
```

**Environment**
- Babel version(s): [e.g. v6.0.0, v7.0.0-beta.34]
- Node/npm version: Node 8.12.0/npm 6.4.1
- OS: Ubuntu 16.04.4 LTS
- Monorepo
- How you are using Babel: [e.g. `cli`, `register`, `loader`]
  cli
  ```
  "build": "babel . --out-dir dist"
  ```

**Comment by nicolo-ribaudo** [comment]
Can you share a small example which reproduces the issue?

**Comment by duzenko** [comment]
```
  test(event) {
    const re = /\p{L}/u
    const f = re.test(event.target.value)
    console.log(f, event.target.value)
  }

```

```
ERROR in ./frontend/App.js
Module build failed (from ./node_modules/babel-loader/lib/index.js):
SyntaxError: Expected atom at position 2
    \p{L}
      ^
    at bail (D:\dev\rnd\webtranslate\node_modules\regjsparser\parser.js:1093:13)
    at parseTerm (D:\dev\rnd\webtranslate\node_modules\regjsparser\parser.js:486:9)
    at parseAlternative (D:\dev\rnd\webtranslate\node_modules\regjsparser\parser.js:457:21)
    at parseDisjunction (D:\dev\rnd\webtranslate\node_modules\regjsparser\parser.js:437:16)
    at parse (D:\dev\rnd\webtranslate\node_modules\regjsparser\parser.js:1108:18)
    at rewritePattern (D:\dev\rnd\webtranslate\node_modules\regexpu-core\rewrite-pattern.js:324:15)
    at PluginPass.RegExpLiteral (D:\dev\rnd\webtranslate\node_modules\@babel\plugin-transform-unicode-regex\lib\index.js:50:51)
    at newFn (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\visitors.js:193:21)
    at NodePath._call (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:53:20)
    at NodePath.call (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:40:17)
    at NodePath.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:88:12)
    at TraversalContext.visitQueue (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:118:16)
    at TraversalContext.visitSingle (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:90:19)
    at TraversalContext.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:146:19)
    at Function.traverse.node (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\index.js:94:17)
    at NodePath.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:95:18)
    at TraversalContext.visitQueue (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:118:16)
    at TraversalContext.visitMultiple (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:85:17)
    at TraversalContext.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:144:19)
    at Function.traverse.node (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\index.js:94:17)
    at NodePath.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:95:18)
    at TraversalContext.visitQueue (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:118:16)
    at TraversalContext.visitMultiple (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:85:17)
    at TraversalContext.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:144:19)
    at Function.traverse.node (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\index.js:94:17)
    at NodePath.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\path\context.js:95:18)
    at TraversalContext.visitQueue (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:118:16)
    at TraversalContext.visitSingle (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:90:19)
    at TraversalContext.visit (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\context.js:146:19)
    at Function.traverse.node (D:\dev\rnd\webtranslate\node_modules\@babel\traverse\lib\index.js:94:17)
 @ ./frontend/index.js 3:0-24 4:36-39
 @ multi ./frontend/index
```

**Comment by AlexSHoffman** [comment]
I have this issue as well, @nicolo-ribaudo 

## Example code:
```JS
"fileName".replace(/(\p{Ll})(\p{Lu})/ug, '$1 $2')
```
- When I run this RegEx command in the console, everything works, I get expected string `"file Name"` back.

- When I run the following RegEx command from within the code, I get the error below:
```JS
Module build failed (from ./node_modules/@vue/cli-plugin-babel/node_modules/babel-loader/lib/index.js):
SyntaxError: Expected atom at position 3
    (\p{Ll})(\p{L
       ^
```

**Environment:** 
Electron app (Chrome 70) w/ Webpack

**Dependencies:**
```
"devDependencies": {
    "@vue/cli-plugin-babel": "^3.4.1",
}
```

**Comment by jridgewell** [comment]
Just enabling the `preset-es2015` causes it... Try checking/unchecking the box in https://babeljs.io/repl/build/master#?babili=false&browsers=Safari%20%3E%3D%208&build=&builtIns=false&spec=false&loose=false&code_lz=EQMwlgNgpgcghgWysAdAJygBwnAxlACgHoCAdTAbwBkIBfASjMqoFcGiWBzAGgAIByACQBGXoIBM_ekA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015&prettier=false&targets=&version=7.3.4

**Comment by r3verser** [comment]
Same error on this regex
`/(?<![\w\d])(aaa|bbb|ccc)(?![\w\d])/iugm;`

> SyntaxError: Expected atom at position 1
    (?<![\w\d])
     ^

**Comment by jasonk** [comment]
So I've managed to narrow this down.  I'm not certain what the "real" fix for it is, but I have some ideas, (and a workaround that is working for me )..

TL;DR - Workaround:
```
[ '@babel/preset-env', {
  include : [ '@babel/plugin-proposal-unicode-property-regex' ],
  exclude : [ '@babel/plugin-transform-unicode-regex' ],
} ],
```

When it fails for me there are two unicode regex related plugins that have been enabled by preset-env: `transform-unicode-regex` and `proposal-unicode-property-regex`.  They are both doing effectively the exact same thing, using `rewritePattern` from `regexpu-core` to transform the regex into an equivalent ES5 regex.

However, only [proposal-unicode-property-regex](https://github.com/babel/babel/blob/1d3f9815df747902ae535706f032081bc4825b87/packages/babel-plugin-proposal-unicode-property-regex/src/index.js#L22) is passing the option `unicodePropertyEscape: true` when it calls `rewritePattern`, [transform-unicode-regex](https://github.com/babel/babel/blob/1d3f9815df747902ae535706f032081bc4825b87/packages/babel-plugin-transform-unicode-regex/src/index.js#L14) is not, which causes [regjsparser](https://github.com/jviereck/regjsparser/blob/0f12db12eb2ecb67143979ddba275c4aa87c40b2/parser.js#L844) to choke on `\p{...}` and `\P{...}` patterns.

Unfortunately I don't understand the Babel internals well enough to determine:
* Why preset-env decided that both of these plugins were needed when the only difference between them is that one of them understands `\p`/`\P` and the other doesn't.
* How it determines what order to apply the plugins in.  Even though they are nearly identical, it seems as though it would work fine if `proposal-unicode-property-regex` were applied first, since it would transform the problematic escapes into something that `transform-unicode-regex` wouldn't have issues with.


If it helps, here is my `preset-env` debug output:
```
@babel/preset-env: `DEBUG` option

Using targets:
{
  "android": "67",
  "chrome": "72",
  "edge": "15",
  "firefox": "60",
  "ie": "11",
  "ios": "11",
  "opera": "46",
  "safari": "11",
  "samsung": "6.2"
}

Using modules transform: false

Using plugins:
  transform-template-literals { "android":"67", "ie":"11", "ios":"11", "safari":"11" }
  transform-literals { "android":"67", "ie":"11" }
  transform-function-name { "android":"67", "edge":"15", "ie":"11" }
  transform-arrow-functions { "android":"67", "ie":"11" }
  transform-block-scoped-functions { "android":"67" }
  transform-classes { "android":"67", "ie":"11" }
  transform-object-super { "android":"67", "ie":"11" }
  transform-shorthand-properties { "android":"67", "ie":"11" }
  transform-duplicate-keys { "android":"67", "ie":"11" }
  transform-computed-properties { "android":"67", "ie":"11" }
  transform-for-of { "android":"67", "ie":"11" }
  transform-sticky-regex { "android":"67", "ie":"11" }
  transform-dotall-regex { "android":"67", "edge":"15", "firefox":"60", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  transform-unicode-regex { "android":"67", "ie":"11", "ios":"11", "safari":"11" }
  transform-spread { "android":"67", "ie":"11" }
  transform-parameters { "android":"67", "edge":"15", "ie":"11" }
  transform-destructuring { "android":"67", "ie":"11" }
  transform-block-scoping { "android":"67", "ie":"11" }
  transform-typeof-symbol { "android":"67", "ie":"11" }
  transform-new-target { "android":"67", "ie":"11" }
  transform-regenerator { "android":"67", "ie":"11" }
  transform-exponentiation-operator { "android":"67", "ie":"11" }
  transform-async-to-generator { "android":"67", "ie":"11" }
  proposal-async-generator-functions { "android":"67", "edge":"15", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  proposal-object-rest-spread { "android":"67", "edge":"15", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  proposal-unicode-property-regex { "android":"67", "edge":"15", "firefox":"60", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  proposal-json-strings { "android":"67", "edge":"15", "firefox":"60", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  proposal-optional-catch-binding { "android":"67", "edge":"15", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
  transform-named-capturing-groups-regex { "android":"67", "edge":"15", "firefox":"60", "ie":"11", "ios":"11", "opera":"46", "safari":"11", "samsung":"6.2" }
```

**Comment by brettz9** [comment]
For named capturing, I had to add both to `exclude`, i.e.:

```json
{
  "presets": [
    ["@babel/preset-env", {
      "exclude": ["@babel/plugin-proposal-unicode-property-regex", "@babel/plugin-transform-unicode-regex"]
    }]
  ]
}
```
